crates/astria-sequencer-relayer/src/sequencer_poller.rs

```diff
@@ -0,0 +1,54 @@
+use std::time::Duration;
+
+use astria_sequencer_client::{
+    BlockResponse,
+    Client,
+};
+use eyre::WrapErr as _;
+use tokio::{
+    sync::mpsc::UnboundedSender,
+    time::interval,
+};
+use tracing::warn;
+
+use crate::config::Config;
+pub(crate) struct SequencerPoller {
+    client: Client,
+    poll_period: Duration,
+    sequencer_blocks_tx: UnboundedSender<BlockResponse>,
+}
+
+impl SequencerPoller {
+    pub(crate) fn new(
+        cfg: &Config,
+        sequencer_blocks_tx: UnboundedSender<BlockResponse>,
+    ) -> eyre::Result<Self> {
+        let client =
+            Client::new(&cfg.sequencer_endpoint).wrap_err("failed to create sequencer client")?;
+        Ok(Self {
+            client,
+            poll_period: Duration::from_millis(cfg.block_time),
+            sequencer_blocks_tx,
+        })
+    }
+
+    pub(crate) async fn run(self) -> eyre::Result<()> {
+        let mut interval = interval(self.poll_period);
+        loop {
+            interval.tick().await;
+            match self.client.get_latest_block().await {
```

Comment:
we should probably change this not to be `get_latest_block` but to be "get all blocks since last height received" as blocks can potentially be missed